webapp/views/App/views/Analysis/Chain/store/calculation/actions/useDismiss.js

```diff
@@ -0,0 +1,55 @@
+import { useDispatch } from 'react-redux'
+
+import { DialogConfirmActions } from '@webapp/store/ui'
+import { AnalysisActions } from '@webapp/service/storage'
+
+import { useReset } from './useReset'
+
+export const useDismiss = ({
+  chain,
+  setChain,
+  step,
+  setStep,
+  setDirty,
+  calculation,
+  originalCalculation,
+  setCalculation,
+  calculationDirty,
+  setCalculationDirty,
+}) => {
+  const dispatch = useDispatch()
+
+  const reset = useReset({
+    chain,
+    setChain,
+    step,
+    setStep,
+    setDirty,
+    calculation,
+    originalCalculation,
+    setCalculation,
+    setCalculationDirty,
+  })
+
+  const resetCalculation = async () => {
+    reset()
+    AnalysisActions.resetCalculation()
+    setCalculation({})
+    setCalculationDirty(false)
```

Comment:
is this state param required? can't you check that calculation !== calculationOriginal ?